lib/Memecoin.ts

```diff
@@ -466,39 +481,67 @@ export class MemecoinSDK {
     return BigInt(result)
   }
 
-  private async estimateSwapCoin(params: EstimateSwapCoinParams): Promise<bigint> {
-    const { fromToken, toToken, amountIn } = params
+  async estimateSwap(params: EstimateSwapParams): Promise<EstimateSwapResult> {
+    const [tokenInMemecoin, tokenOutMemecoin] = await Promise.all([
+      this.getCoin(params.tokenIn),
+      this.getCoin(params.tokenOut)
+    ])
 
-    const response = await fetch(`${this.apiBaseUrl}/api/trades/estimate-swap`, {
-      method: 'POST',
-      headers: {
-        'Content-Type': 'application/json'
-      },
-      body: JSON.stringify({
-        fromToken,
-        toToken,
-        amountIn: amountIn.toString()
-      })
-    })
+    const tokenInData = tokenInMemecoin
+      ? this.determinePoolOfMemecoin(tokenInMemecoin)
+      : await this.findTokenWETHPool(params.tokenIn)
+    const tokenOutData = tokenOutMemecoin
+      ? this.determinePoolOfMemecoin(tokenOutMemecoin)
+      : await this.findTokenWETHPool(params.tokenOut)
```

Comment:
add an LRU cache for at least 1 min or something. we're going to call estimateSwap on every keyboard keystroke

cache value should be a promise so prevent multiple calls to server i.e. `const poolCache = LRUCache(EthAddress, Promise<GetTokenPoolResponse>`